### ISSUE TITLE
DBA Dash error for Source IdentityColumns

### ISSUE THREAD

**Issue by VincenzoMarchese** [trimble-oss/dba-dash#962]
DBA Dash is reporting errors for the source IdentityColumns, error context Collect[Retrying] 

Microsoft.Data.SqlClient.SqlException (0x80131904): **Cannot insert the value NULL into column 'row_count', table 'DBADashDB.dbo.IdentityColumnsHistory'; column does not allow nulls. INSERT fails.
The statement has been terminated.**
```
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, SqlCommand command, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at DBADash.DBImporter.Update(String tableName) in D:\a\dba-dash\dba-dash\DBADash\DBImporter.cs:line 513
   at DBADash.DBImporter.<>c__DisplayClass17_0.<TryUpdate>b__0(Context context) in D:\a\dba-dash\dba-dash\DBADash\DBImporter.cs:line 463
   at Polly.Policy.<>c__DisplayClass138_0.<Implementation>b__0(Context ctx, CancellationToken token)
   at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Action`4 onRetry, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider)
ClientConnectionId:50b15b46-39a9-4383-b565-9f018570ccdd
Error Number:515,State:2,Class:16```

**Comment by DavidWiseman** [comment]
The row count comes from sys.dm_db_partition_stats.  This is nullable and it's added in with an outer apply.  A null value is possible and should be handled - this will need to be looked at and fixed in a future version.  It is a little odd that you are getting a NULL value though.  

If you wanted to check you could run [this query](https://raw.githubusercontent.com/trimble-oss/dba-dash/main/DBADash/SQL/SQLIdentityColumns.sql) directly against your instance.  You will need to add `DECLARE @IdentityCollectionThreshold  INT =5` to the top of the script as the application passes in this parameter.  You should be able to see which table has the NULL value.  You might be able to do some investigations on why that is the case.

**Comment by DavidWiseman** [comment]
This will be fixed in the next release.

**Comment by DavidWiseman** [comment]
Included in 3.8.0 🎉
